### PR TITLE
feat(mme): Support for PCO with DNS option in amf

### PR DESCRIPTION
**Pull request by panyogesh**
Fix:
  Enable the encoding/decoding of the options in NAS5G
  Enabling the DNS Server configuration for AMF

Test:
  Unit testing
  Simulator testing

Logs:
  10/13 Test #10: test_amf .........................   Passed    0.06 sec

Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h [hunk 1]
```diff
@@ -886,6 +886,11 @@ int decode_access_point_name_ie(
 /* Both directions:*/
 #define PCO_CI_IM_CN_SUBSYSTEM_SIGNALING_FLAG (0x0002)
 
+/* IPCP Macros */
+#define IPCP_CODE_BYTES 1
+#define IPCP_IDENTIFIER_BYTES 1
+#define IPC_LENGTH_BYTES 2
+
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h]: Not sure IPCP_* should go into 24.008 related .h file.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp [hunk 2]
```diff
@@ -812,13 +814,14 @@ int amf_app_handle_pdu_session_accept(
...
-  encode_msg->payload_container.len = 30;
-  len                               = 41;  // originally 38 and 30
+  msg_accept_pco =
+      &(smf_msg->msg.pdu_session_estab_accept.protocolconfigurationoptions.pco);
+
+  auto pco_len = sm_process_pco_request(&(smf_ctx->pco), msg_accept_pco);
+
+  encode_msg->payload_container.len =
+      PDU_ESTAB_ACCPET_PAYLOAD_CONTAINER_LEN + pco_len;
+  len = PDU_ESTAB_ACCPET_NAS_PDU_LEN + pco_len;
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: typo accept

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context_and_proc.h [hunk 3]
```diff
@@ -95,7 +95,7 @@ struct amf_procedures_t;
 #define N1_SM_INFO 0x1
 #define AMBR_LEN 6
 #define PDU_ESTAB_ACCPET_PAYLOAD_CONTAINER_LEN 30
-#define PDU_ESTAB_ACCPET_NAS_PDU_LEN 39
+#define PDU_ESTAB_ACCPET_NAS_PDU_LEN 41
```
Comment (lionelgo) [inline lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context_and_proc.h]: Typo accept ?

File: lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h (hunk 1 above)
Comment (panyogesh) [reply lte/gateway/c/core/oai/lib/3gpp/3gpp_24.008.h]: Comments addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp (hunk 2 above)
Comment (panyogesh) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp]: Comments addressed.

File: lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context_and_proc.h (hunk 3 above)
Comment (panyogesh) [reply lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context_and_proc.h]: Comments addressed.

**Comment by panyogesh** [comment]
image

**Review by ulaskozat — CHANGES_REQUESTED** [review]
conflict exists

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**